proposals/filecoin-pubsub-observatory.md

```diff
@@ -0,0 +1,82 @@
+# Filecoin PubSub Observatory
+
+Authors: vyzo
+
+Initial PR: TBD <!-- Reference the PR first proposing this document. Oooh, self-reference! -->
+
+## Purpose &amp; impact
+#### Background &amp; intent
+
+We currently have little visibility in terms of block propagation and
+network health in the filecoin network.  We are also lacking a
+reputation system for filecoin peers.
+
+Investigating and debugging apparent block propagation problems is
+difficult without observability into the network’s pubsub activity,
+such as message flows, peer scores, and low-level RPCs (that can be
+processed to extract higher-level insights).
+
+
+#### Assumptions &amp; hypotheses
+- We care about block propagation and pubsub peer scores in the filecoin network.
+
+#### User workflow example
+
+- A lotus dev (or external party if we make the relevant dashboards open)
+  could trace block propagation through the network
+- A node operator could check on his nodes score and eventually reputation.
+
+
+#### Impact
+
+The project would allow us to gain insights in the filecoin network,
+which could be invaluable in understanding, debugging, and improving
+the system.
+
+#### Leverage
+Quite decent.
+
+#### Confidence
+Pretty good.
+
+
+## Project definition
+#### Brief plan of attack
+
+1. To evolve the current pubsub traces collector (which receives
+   traces from 1.5k concurrent Lotus nodes) to ship the data stream to
+   our logz.io Elasticsearch, where it can be queried with Kibana to
+   explore and build dashboards.
```

Comment:
cc @lanzafame  as well as he thinks about the moving parts here. adding this stream shouldn't be bad.

@vyzo - can you clarify if the data stream at the trace collector is semi-structured logs or structured data of observed peers / scores?

Reply:
It's fully structured, we collect protobufs (and can also emit json in the trace collector).
Here is the pb definition: https://github.com/libp2p/go-libp2p-pubsub/blob/master/pb/trace.proto

For the scores and statistics we don't currently do any collection, this will have to be added to lotus (or possibly a side process).